bolt/lib/linear_solver/EM_fields_solver.py

```diff
@@ -5,23 +5,26 @@
 
 def compute_electrostatic_fields(self):
     """
-    Computes the electrostatic fields by making use of the Poisson equation:
-    div^2 phi = rho
+    Computes the electrostatic fields by making use of the
+    Poisson equation: div^2 phi = rho
     """
     rho_hat = 2 * af.fft2(self.compute_moments('density'))/\
               (self.N_q1 * self.N_q2) # Scaling Appropriately
 
-    rho_hat = af.tile(rho_hat, 1, 1, self.N_p1 * self.N_p2 * self.N_p3)
     phi_hat = self.physical_system.params.charge_electron * \
               af.broadcast(lambda a, b:a/b, rho_hat, 
```

Comment:
Can we avoid using lambda functions?

Reply:
Do you suggest that each function has its own set of primitive operations using lambda functions, or do we have a file outside called `utils.py` or something where everything is defined? We can probably add things like the bandwidth test to this file as well.

Reply:
Yes, I prefer have `utils.py` and reusing the functions rather then code duplication all over.